platform/admin/configurator/admin-configurator-api/src/main/java/org/codice/ddf/admin/configurator/Configurator.java

```diff
@@ -0,0 +1,204 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+/**
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface Configurator {
+    /**
+     * Sequentially invokes all the transaction's operations, committing their changes. If a failure
+     * occurs during the processing, a rollback is attempted of those handlers that had already been
+     * committed.
+     * <p>
+     * In the case of a successful commit, changes should be logged.
+     *
+     * @param auditMessage In the case of a successful commit, the message to pass to a
+     *                     {@code SecurityLogger}
+     * @param auditParams  In the case of a successful commit, the optional parameters to pass to a
+     *                     {@code SecurityLogger} to be interpolated into the message
+     * @return report of the commit status, whether successful, successfully rolled back, or partially
+     * rolled back with errors
+     */
+    OperationReport commit(String auditMessage, String... auditParams);
+
+    /**
+     * Sequentially invokes all the transaction's operations, committing their changes. If a failure
+     * occurs during the processing, a rollback is attempted of those handlers that had already been
+     * committed.
+     *
+     * @return report of the commit status, whether successful, successfully rolled back, or partially
+     * rolled back with errors
+     */
+    OperationReport commit();
+
+    /**
+     * Starts the bundle with the given name.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String startBundle(String bundleSymName);
+
+    /**
+     * Stops the bundle with the given name.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return a lookup key that can be used to correlate this operation in the
+     * final {@link OperationReport}
+     */
+    String stopBundle(String bundleSymName);
+
+    /**
+     * Determines if the bundle with the given name is started.
+     *
+     * @param bundleSymName the symbolic name of the bundle
+     * @return true if started; else, false
+     */
+    boolean isBundleStarted(String bundleSymName);
```

Comment:
❓These "read" methods (`isBundleStarted`, `isFeatureStarted`, `getProperties`, `getConfig`, `getManagedServiceConfigs` and `getServiceReference`) do not fit very well with the other concepts exposed by this interface (transactions, operations, commit/rollback). For instance, would calling the `isBundleStarted("bundle")` after calling `startBundle("bundle")` but before calling `commit()` return `true` or `false`?

To make this a little clearer, could we move them to separate interfaces? For instance, could we define something like a `BundleController` interface that allows a client to start, stop and check the status of a bundle, `StartBundleOperation` and `StopBundleOperation` interfaces and classes that use the `BundleController` when `commit` is called and know how to rollback if something fails?